File: QuasarCode_Library/Games/Cards/PlayingCard.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Linq;

namespace QuasarCode.Library.Games.Cards
{
    /// <summary>
    /// A card from a standard deck of 52 playing cards
    /// </summary>
    public class PlayingCard : IPlayingCard
    {
        /// <summary>
        /// The card's value
        /// </summary>
        public AllowedValues Value { get; }

        /// <summary>
        /// The card's suit
        /// </summary>
        public AllowedSuits Suit { get; }
        
        /// <summary>
        /// Creates a new PlayingCard instance
        /// </summary>
        /// <param name="value">The card's value</param>
        /// <param name="suit">The card's suit</param>
        public PlayingCard(AllowedValues value, AllowedSuits suit)
        {
            Value = value;

            Suit = suit;
        }



        private static readonly string[] ValueStrings = new string[] { "A", "2", "3", "4", "5", "6", "7", "8", "9", "10", "J", "Q", "K" };

        /// <summary>
        /// Enum of card values from ace to king
        /// </summary>
        public enum AllowedValues
        {
            /// <summary>Ace</summary>
            A,

            /// <summary>Two</summary>
            Two,

            /// <summary>Three</summary>
            Three,

            /// <summary>Four</summary>
            Four,

            /// <summary>Five</summary>
            Five,

            /// <summary>Six</summary>
            Six,

            /// <summary>Seven</summary>
            Seven,

            /// <summary>Eight</summary>
            Eight,

            /// <summary>Nine</summary>
            Nine,

            /// <summary>Ten</summary>
            Ten,

            /// <summary>Jack</summary>
            J,

            /// <summary>Queen</summary>
            Q,

            /// <summary>King</summary>
            K,

            /// <summary>Joker</summary>
            Jo
        }

        /// <summary>
        /// Retrive the string counterpart to the AllowedValues enum value
        /// </summary>
        /// <param name="value">Card value from AllowedValues</param>
        /// <returns>String representing the provided value</returns>
        public static string GetValueString(AllowedValues value)
        {
            return ValueStrings[(int)value];
        }


        private static readonly string[] SuitStrings = new string[] { "Heart", "Club", "Dimond", "Spade", "Joker" };

        /// <summary>
        /// Enum of card suits - hearts, clubs, dimonds, spades, joker
        /// </summary>
        public enum AllowedSuits
        {
            /// <summary>Heart</summary>
            H,

            /// <summary>Club</summary>
            C,

            /// <summary>Dimond</summary>
            D,

            /// <summary>Spade</summary>
            S,

            /// <summary>Joker</summary>
            J
        }

        /// <summary>
        /// Retrive the string counterpart to the AllowedSuits enum suit
        /// </summary>
        /// <param name="suit">Card suit from AllowedSuits</param>
        /// <returns>String representing the provided suit</returns>
        public static string GetSuitString(AllowedSuits suit)
        {
            return SuitStrings[(int)suit];
        }
    }
}

File: QuasarCode_Python/QuasarCode/Games/Cards/_PlayingCard.py
from QuasarCode.Games.Cards import IPlayingCard
from enum import Enum

class PlayingCard(IPlayingCard):
    """
    A card from a standard deck of 52 playing cards

    Atributes:
        PlayingCard.AllowedValues Value --> The card's value
        PlayingCard.AllowedSuits Suit --> The card's suit
    """

    class AllowedValues(Enum):
        """
        Enum of card values from ace to king and joker
        """
        
        A = 0,# Ace

        Two = 1,# Two

        Three = 2,# Three

        Four = 3,# Four

        Five = 4,# Five

        Six = 5,# Six

        Seven = 6,# Seven

        Eight = 7,# Eight

        Nine = 8,# Nine

        Ten = 9,# Ten

        J = 10,# Jack

        Q = 11,# Queen

        K = 12,# King

        Jo = 13# Joker

    class AllowedSuits(Enum):
        """
        Enum of card suits - hearts, clubs, dimonds, spades, joker
        """
        
        H = 0,# Heart

        C = 1,# Club

        D = 2,# Dimond

        S = 3,# Spade

        J = 4# Joker

    def __init__(self, value: AllowedValues, suit: AllowedSuits):
        self.__value = value
        self.__suit = suit

    @property
    def Value(self) -> AllowedValues:
        """
        The card's value
        """
        return self.__value

    @property
    def Suit(self) -> AllowedSuits:
        """
        The card's suit
        """
        return self.__suit

    __ValueStrings = ("A", "2", "3", "4", "5", "6", "7", "8", "9", "10", "J", "Q", "K")

    def GetValueString(value: AllowedValues) -> str:
        """
        Retrive the string counterpart to the AllowedValues enum value
        """
        return PlayingCard.__ValueStrings[int(value)]

    __SuitStrings = ("Heart", "Club", "Dimond", "Spade", "Joker")

    def GetSuitString(suit: AllowedSuits) -> str:
        """
        Retrive the string counterpart to the AllowedSuits enum suit
        """
        return PlayingCard.__SuitStrings[int(suit)]